### PR TITLE
Add support to multiple event Azure service bus

### LINKED ISSUE #28: Add support to multiple event buses
Client should be able to use multiple event buses at the same time of any tech

### PR DESCRIPTION
**Pull request by Krusty93**
Fixes #28 

@matteopessina I created a draft PR with a PoC of one of the possible solutions that would enable ebe to support multiple azure service bus at the same time.

Long story short, you need to specify in each request as HEADER the name of the service bus you want to operate on. If only one service bus is attached, the header parameter is not necessary; if two or more service bus are attached and the HEADER is empty (or value doesn't match with the config) an error is thrown.

I also added some comments to highlight some parts

### CODE REVIEW

File: src/Server/EventBusExplorer.Server.API/Middlewares/HeaderValidatorMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        await next(context);
+    }
+}
+
+public static class HeaderValidatorMiddlewareExtensions
+{
+    public static IApplicationBuilder UseHeaderValidator(this IApplicationBuilder builder) =>
+        builder.UseWhen(context =>
+            !context.Request.Path.StartsWithSegments("/swagger"),
```
Comment (krusty93) [inline src/Server/EventBusExplorer.Server.API/Middlewares/HeaderValidatorMiddleware.cs:75]: I'd prefer to invert the condition (apply the filter only if querying APIs). It'd be possible if a prefix like `/api` would be used as prefix for all APIs, not now

File: src/Server/EventBusExplorer.Server.API/Swagger/SwaggerBusNameHeader.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.OpenApi.Models;
```
Comment (krusty93) [inline src/Server/EventBusExplorer.Server.API/Swagger/SwaggerBusNameHeader.cs:1]: this file can be deleted if controllers' actions have `[FromHeader]`

File: src/Server/EventBusExplorer.Server.Infrastructure.AzureServiceBus/Bootstrapper.cs [hunk 3]
```diff
@@ -24,11 +28,61 @@ public static class Bootstrapper
...
+                    {
+                        if (dict.Count == 1)
+                        {
+                            cs = dict.FirstOrDefault().Value;
+                        }
+                    }
+
+                    return new ServiceBusAdministrationClient(cs);
+                };
+            });
```
Comment (krusty93) [inline src/Server/EventBusExplorer.Server.Infrastructure.AzureServiceBus/Bootstrapper.cs:73]: This code will be refactored

File: src/Server/EventBusExplorer.Server.Infrastructure.AzureServiceBus/Bootstrapper.cs [hunk 4]
```diff
@@ -24,11 +28,61 @@ public static class Bootstrapper
...
+{
+    public IDictionary<string, string> AzureServiceBus { get; init; } = new Dictionary<string, string>();
+
+    public ReadOnlyCollection<string> ServiceBusNames =>
+        AzureServiceBus
+            .Keys
+            .OrderBy(x => x)
+            .ToList()
+            .AsReadOnly();
+}
```
Comment (krusty93) [inline src/Server/EventBusExplorer.Server.Infrastructure.AzureServiceBus/Bootstrapper.cs:88]: Settings have this schema:
```json
  "ConnectionStrings": {
    "EventBus": {
      "AzureServiceBus": {
        "name1": "cs1",
        "name2": "cs2"
      }
    }
  },
```

File: src/Server/EventBusExplorer.Server.Infrastructure.AzureServiceBus/ServiceBusQueueService.cs [hunk 5]
```diff
@@ -11,15 +11,21 @@ internal class ServiceBusQueuesService : IServiceBrokerQueuesService
...
-        _client = client ??
+        _client = client.Invoke("Platform") ??
             throw new ArgumentNullException(nameof(client));
 
-        _adminClient = adminClient ??
+        _adminClient = adminClient.Invoke("Platform") ??
             throw new ArgumentNullException(nameof(adminClient));
+
+        _clientDelegate = client;
+        _adminClientDelegate = adminClient;
```
Comment (krusty93) [inline src/Server/EventBusExplorer.Server.Infrastructure.AzureServiceBus/ServiceBusQueueService.cs:28]: Of course this shall be cleaned up

**Comment by Krusty93** [comment]
README.md file needs to get updated with this PR

**Pull request closed without merging**